astropy/units/decorators.py

```diff
@@ -127,7 +138,11 @@ def wrapper(*func_args, **func_kwargs):
 
             # Call the original function with any equivalencies in force.
             with add_enabled_equivalencies(self.equivalencies):
-                return wrapped_function(*func_args, **func_kwargs)
+                return_ = wrapped_function(*func_args, **func_kwargs)
+            if wrapped_signature.return_annotation is not funcsigs._empty:
```

Comment:
https://docs.python.org/3/library/inspect.html suggests that there is a non-private `Signature.empty` property. Similarly, for funcsigs: http://funcsigs.readthedocs.io/en/0.4/